ebean-core/src/test/java/org/tests/model/m2m/MnyNode.java

```diff
@@ -0,0 +1,129 @@
+package org.tests.model.m2m;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+
+import io.ebean.annotation.Platform;
+import io.ebean.annotation.Where;
+
+import java.util.List;
+
+@Entity
+public class MnyNode {
+
+  @Id
+  Integer id;
+
+  String name;
+
+  @ManyToMany
+  @JoinTable(name = "mny_edge",
+    joinColumns = @JoinColumn(name = "from_id", referencedColumnName = "id"),
+    inverseJoinColumns = @JoinColumn(name = "to_id", referencedColumnName = "id"))
+  List<MnyNode> allRelations;
+
+  @ManyToMany
+  @JoinTable(name = "mny_edge",
+    joinColumns = @JoinColumn(name = "to_id", referencedColumnName = "id"),
+    inverseJoinColumns = @JoinColumn(name = "from_id", referencedColumnName = "id"))
+  List<MnyNode> allReverseRelations;
+
+  @ManyToMany
+  @JoinTable(name = "mny_edge",
+    joinColumns = @JoinColumn(name = "from_id", referencedColumnName = "id"),
+    inverseJoinColumns = @JoinColumn(name = "to_id", referencedColumnName = "id"))
+  @Where(clause = "${mta}.flags & 1 != 0")
+  @Where(clause = "BITAND(${mta}.flags, 1) != 0", platforms = Platform.H2)
+  List<MnyNode> bit1Relations;
```

Comment:
This entity demonstrates how to use an existing entity as ManyToMany table and add certain where filters.

Reply:
excellent